main.py

```diff
@@ -110,61 +129,70 @@ def upload_file(
         f"as {object_name or file_name}"
     )
     # If S3 object_name was not specified, use file_name
-    if object_name is None:
+    if not object_name:
         object_name = file_name
 
     # Upload the file
-    s3_client = boto3.client('s3',
-                             region_name="us-east-1",
-                             aws_access_key_id=ACCESS_KEY,
-                             aws_secret_access_key=SECRET_KEY)
+    s3_client = boto3.client(
+        service_name="s3",
+        region_name="us-east-1",
+        aws_access_key_id=ACCESS_KEY,
+        aws_secret_access_key=SECRET_KEY
+    )
     try:
-        s3_client.upload_file(file_name, bucket, object_name,
-                              ExtraArgs={'ContentType': content_type})
+        s3_client.upload_file(
+            file_name,
+            bucket,
+            object_name,
+            ExtraArgs={"ContentType": content_type}
+        )
     except ClientError as e:
         logging.error(e)
         return False
     return True
 
 
 def upload_coverage_file(
-        coverage_percentage,
-        badge_name,
-        bucket
-):
+    coverage_percentage: int,
+    badge_name: str,
+    bucket: str
+) -> bool:
     """
     Upload a file with the coverage percentage to an S3 bucket.
 
     :param int coverage_percentage: Current coverage  percentage
     :param str badge_name: Name of the badge file
     :param str bucket: Bucket to upload to
-    :param str|None object_name: S3 object name. If not specified then
-        file_name is used
     :return: True if file was uploaded, else False
     :rtype: bool
     """
     coverage_file = os.path.splitext(badge_name)[0] + ".txt"
-    temp_coverage_file = "coverage-percentage.tx"
+    temp_coverage_file = "coverage-percentage.txt"
     logging.info(
-        f"Writing coverage percentage to file {coverage_file} to {bucket} ."
+        f"Writing coverage percentage to file {coverage_file} to {bucket}."
     )
-    with open(temp_coverage_file, 'w') as f:
-        logging.info(f"Coverage  : {coverage_percentage}")
+    with open(temp_coverage_file, "w") as f:
+        logging.info(f"Coverage: {coverage_percentage}")
         f.write(str(coverage_percentage))
 
     # Upload the file
-    upload_file(temp_coverage_file, bucket, "text/plain", coverage_file)
+    return upload_file(
+        file_name=temp_coverage_file,
+        bucket=bucket,
+        content_type="text/plain",
+        object_name=coverage_file
+    )
 
 
-def main():
+def main() -> None:
     """
     Parse coverage file, create appropriate badge and upload to S3.
     """
-    coverage = compute_coverage()
+    coverage = get_coverage()
```

Comment:
We should add a check here in order to validate if one of `coverage_percentage_regex`, `coverage_json` vars has been set